src/main/java/org/alfresco/repo/copy/CopyServiceImpl.java

```diff
@@ -1519,4 +1520,22 @@ public String getTopLevelNodeNewName(NodeRef sourceNodeRef, NodeRef targetParent
         return null;
     }
 
+    /**
+     * Builds name by appending copy label.
+     */
+    String buildNewName(NodeRef destinationParent, QName assocTypeQName, String newName)
+    {
+        String baseName = FilenameUtils.getBaseName(newName);
+        String extension = FilenameUtils.getExtension(newName);
+
+        newName = I18NUtil.getMessage(COPY_OF_LABEL, baseName);
+
+        // append extension, if any, to filename
+        if (extension != null && !extension.isEmpty())
+        {
+            newName = newName + FilenameUtils.EXTENSION_SEPARATOR_STR + extension;
+        }
+
+        return newName;

```

Comment:
You are modifying the name using the parameter variable newName. You should use a new variable name and treat String newName as though it was declared final. It might be better to change the parameter to inputName or just name.